### ISSUE TITLE
Docker image

### ISSUE THREAD

**Issue by LoveIsGrief** [zlatinb/muwire#32]
It would be nice to have a docker image / Dockerfile to make this project available on [DockerHub](https://hub.docker.com/).

With the correct options it would be possible to contain MuWire and easily distribute it. Once work is done on the WebUI it would also be perfect to run on a server side-by-side with I2P.

**Comment by zlatinb** [comment]
Hey @LoveIsGrief I have implemented a setup wizard in MuWire 0.7.1.  The current docker image afaik ships with it's own `MuWire.properties`; if that file exists the setup wizard will not run and will not ask the user for a nickname.  As a result, anyone running the docker image will end up with the default nickname "MuWireUser".

Is it possible for you to take a look into removing the bundled MuWire.properties from the docker image?

**Comment by LoveIsGrief** [comment]
Hi @zlatinb,

The bundles `MuWire.properties` is necessary for pointing to the correct [`downloadLocation`](https://github.com/zlatinb/muwire/blob/master/docker/rootfs/muwire/.MuWire/MuWire.properties#L7) and `incompleteLocation`. Is it possible for the wizard to read default values from a file / point it to a file with default values? Then it would be possible to point to a different default file in the docker image.

**Comment by zlatinb** [comment]
Ok, I'll add a `WizardDefaults` class which will contain defaults for all settings that the wizard can set.  You will need to pass `-Dwizard.defaults=/path/to/defaults.properties` from the Docker start script.
While I do that, pls register at https://i2pgit.org (or inside I2P http://git.idk.i2p) because the GitHub repo is now a read-only mirror to that GitLab repo.

**Comment by zlatinb** [comment]
Done. The class is `gui/src/main/groovy/com/muwire/gui/wizard/WizardDefaults.groovy` and gets initialized from a `java.util.Properties` object.  You can change all defaults except the Nickname.  See associated commit above for details.

**Comment by zlatinb** [comment]
Documented the wizard defaults on the wiki: https://github.com/zlatinb/muwire/wiki/Packaging
Marking this issue as needing help since I'm afraid to touch Docker :-/

**Comment by LoveIsGrief** [comment]
Damn, I'm sorry. I totally forgot about this. To be honest, I wasn't too keen on creating yet another account on yet another gitlab instance. Hopefully they'll introduce federation sometime in the future...


I made an MR https://i2pgit.org/zlatinb/muwire/-/merge_requests/49

**Comment by zlatinb** [comment]
Thanks, it should be possible to sign in with your github account, check with @eyedeekay (he runs the gitlab instance)

I've sent you an invite to become a collaborator on github.  It's a bit of a mess right now - the code is on GL but I'm using the GH wiki (too many links to update).

**Comment by LoveIsGrief** [comment]
OK, thank you :+1:  I'll try and update the docker wiki

**Comment by zlatinb** [comment]
Thanks.  See my other comment in GL about how to set the default I2CP interface and port.  Also, do you want to disable in-network updates in the docker image?

**Comment by eyedeekay** [comment]
re: github wiki, I have the same problem, "Find or create github wiki->gitlab migration tools" is on my list. Creating an account with github should work. [i2pgit.org|git.idk.i2p]/idk/git-over-i2p is the repo where I'm keeping track of git related instructions, if you need to, use it to file issues concerning the configuration or features of the gitlab instance.

**Comment by LoveIsGrief** [comment]
@eyedeekay I saw the github login after I created my account so I can't tell if it works or not :yum: 

@zlatinb If in-network updates update muwire within the container, it might make sense to disable them and keep muwire the same version within the docker container. That would allow the user to manually download a new image for upgrades. Is there some kind of notification that an update is available?

I also think we can close this issue and create new ones when we encounter problems with muwire in docker.

**Comment by zlatinb** [comment]
The in-network update is very primitive - it only downloads the new .zip distribution to the downloads folder and notifies the user that it has been downloaded.  The user then has to manually unzip and deploy the update, so I imagine that makes it useless for docker users.  There is no way to disable the download but leave the notification functionality unless the user does that in the options.

Closing this issue for now, can open a separate one for the updates if desired.  Thanks for your help!
